### PR TITLE
Bug 1753778: report Available when registry is explicitly Removed

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

**Comment by dmage** [comment]
@bparees yes, I want to backport the CDR from master next week.

File: test/e2e/managementstate_test.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// TODO(dmage): wait for the resource to be observed
```
Comment (ricardomaraschini) [inline test/e2e/managementstate_test.go]: Are we planning to wait for this one before merging? This could generate some flake events, no?

Comment (dmage) [reply test/e2e/managementstate_test.go]: No, we have retries below.

File: pkg/operator/status.go [hunk 2]
```diff
@@ -87,10 +87,18 @@ func (c *Controller) syncStatus(cr *imageregistryv1.Config, deploy *appsapi.Depl
 		Message: "",
 		Reason:  "",
 	}
-	if deploy == nil {
+	if cr.Spec.ManagementState == operatorapiv1.Unmanaged {
+		operatorAvailable.Status = operatorapiv1.ConditionTrue
```
[the remark this answers is not in the document]
Comment (dmage) [reply pkg/operator/status.go]: Yes, but we don't want to [block upgrades](https://github.com/openshift/cluster-version-operator/blob/a17c59a1fd6c03f9af40082c380888523d60820b/pkg/cvo/internal/operatorstatus.go#L149-L189), so we report that everything is OK.

[the remark this answers is not in the document]
Comment (adambkaplan) [reply pkg/operator/status.go]: @dmage Think of it this way - if not in the `Managed` state, `Available` means "Can the _operator_ do what you ask it to do?" Unless we are not running, the answer is yes/`True`.

Ex: if the management state is changed from `Unmanaged` to `Managed`, will the operator react as expected? Yes.

**Comment by adambkaplan** [comment]
/hold

Known issue with 1.16 storage tests (waiting for kubelet/RHCOS skew to be reconciled).

File: pkg/operator/status.go [hunk 3]
```diff
@@ -87,10 +87,18 @@ func (c *Controller) syncStatus(cr *imageregistryv1.Config, deploy *appsapi.Depl
...
+		operatorAvailable.Message = "The registry configuration is set to unmanaged mode"
+		operatorAvailable.Reason = "Unmanaged"
+	} else if deploy == nil {
 		if e, ok := applyError.(permanentError); ok {
 			operatorAvailable.Message = applyError.Error()
 			operatorAvailable.Reason = e.Reason
+		} else if removed {
+			operatorAvailable.Status = operatorapiv1.ConditionTrue
+			operatorAvailable.Message = "The deployment is removed"
```
Comment (ricardomaraschini) [inline pkg/operator/status.go]: Could we make this Message something like down below when we read:

"The registry is removed"

File: test/e2e/managementstate_test.go (hunk 1 above)
Comment (ricardomaraschini) [reply test/e2e/managementstate_test.go]: So we don't need the TODO right? Or I misunderstand you?

File: test/e2e/managementstate_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+	err = client.Deployments(imageregistryv1.ImageRegistryOperatorNamespace).Delete(imageregistryv1.ImageRegistryName, &metav1.DeleteOptions{})
```
Comment (ricardomaraschini) [inline test/e2e/managementstate_test.go]: Isn't this Delete automatically happen slightly after we set the Management State to Removed?

**Comment by adambkaplan** [comment]
/lgtm cancel

@dmage please address or remove the TODOs in your comments.

/retest

**Comment by adambkaplan** [comment]
/hold cancel

RHCOS skew issues should be resolved now

**Pull request merged**